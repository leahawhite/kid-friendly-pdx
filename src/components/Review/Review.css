.places-item-review {
  padding: 10px 0;
  border-bottom: 1px solid gray;
  list-style: none;
}

.places-item-review-star-rating {
  float: left;
}

.places-item-review-star-rating span {
  opacity: 0.7;
}

.username {
  font-weight: bold;
}

.review-text {
  clear: both;
  margin: 15px 0;
}

.review-image-container {
  width: auto;
  display: flex;
  margin: 10px 0;
}

.read-btn {
  background-color: #b3f0ff;
  color:#444;
  padding: 0;
  border: 0;
  font-family: 'Source Sans Pro';
  font-weight: bold;
  margin-left: 5px;
}