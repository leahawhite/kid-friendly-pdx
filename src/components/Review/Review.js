import React, { Component } from 'react';
import { StarRating } from '../../components/StarRating/StarRating';
import CarouselLB from '../../components/CarouselLB/CarouselLB';
import moment from 'moment';
import './Review.css';

export default class Review extends Component {
  static defaultProps = { review: {}, users: [] }

  constructor(props) {
    super(props)
    this.state = {
      expanded: false,
    }
  }

  handleExpandText = () => {
    this.setState({
      expanded: !this.state.expanded,
    });
  }

  getTruncatedText(review, words) {
    if (!this.state.expanded) {
      return (
        <>
        {words.slice(0, 25).join(' ') + ' ...'}
        <button className="read-btn" onClick={this.handleExpandText}>Read more</button>
        </>
      )
    } else {
      return (
        <>
        {review.text}
        <button className="read-btn" onClick={this.handleExpandText}>Read less</button>
        </>
      )
    }
  }
  
  getReviewContent = (review) => {
    const words = review.text.split(' ')
    if (words.length < 25) {
      return review.text
    } else {
      return this.getTruncatedText(review, words)
    }
  }

  render() {
    const { review } = this.props
      return (
        <li className="places-item-review" key={review.id}>
          <p className="username">{review.user.display_name}</p>
          <div className="places-item-review-star-rating">
            <div className="star-rating">
            <StarRating rating={review.rating} />
            </div>
            <span>{moment(new Date(review.date_created)).format('MM/DD/YYYY')}</span>
          </div>
          <div className="review-content">
            {this.getReviewContent(review)}
          </div>
          <div className="review-image-container">
            {review.images && review.images.length ?    
            <CarouselLB images={review.images} imagesClass='review-image'/>
            : null}
            </div>
        </li>
      )
  }
}
