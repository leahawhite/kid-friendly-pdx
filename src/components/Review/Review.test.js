import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Review from './Review';

const testReview = { 
  id: 1,
  star_rating: 5,
  text: "Atlas is our new favorite spot for a quick, painless dinner with our toddler. It's super casual -- order slices or pies at the counter, and they call your name for pickup when it's ready. Since it's new, the play area is pretty clean and well-stocked. There's an old church pew there for parents to sit and supervise, but the open floor plan means that you can see the play area from pretty much every table. We've only ordered slices, so we've never had to wait long. We're usually in and out in 30 minutes, which is just about perfect for us.",
  date_created: "2019-06-13",
  place_id: 1,
  user_id: 5,
  images: [
    {
      id: 1,
      src: 'https://lh3.googleusercontent.com/QDwk6G6ZCwYzVH-yJHQ824GZ1wKW2XXus8hxxdH-4AAHz-FejC-UVydyE5le1ZvPTIo1cXw1G25qhljwS8dgyIedCCFtIxBonznVQ4LfaPDEZVWmqKS1vFqVUFFNmMmMlKQ5eCpfhNI=w2400',
      title: `Atlas Pizza's play area`,
      place_id: 1,
      user_id: 5,
    },
  ]
};

describe('Review component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Review />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders UI as expected with props', () => {
    const wrapper = shallow(<Review review={testReview} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})