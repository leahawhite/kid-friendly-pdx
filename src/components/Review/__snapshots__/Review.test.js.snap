// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component renders UI as expected with props 1`] = `
<li
  className="places-item-review"
  key="1"
>
  <p
    className="username"
  >
    0
  </p>
  <div
    className="places-item-review-star-rating"
  >
    <div
      className="star-rating"
    >
      <StarRating
        rating={5}
      />
    </div>
    <span>
      06/13/2019
    </span>
  </div>
  <div
    className="review-text"
  >
    Atlas is our new favorite spot for a quick, painless dinner with our toddler. It's super casual -- order slices or pies at the counter, ...
    <button
      className="read-btn"
      onClick={[Function]}
    >
      Read more
    </button>
  </div>
  <div
    className="review-image-container"
  >
    <LightboxExample
      images={
        Array [
          Object {
            "id": 1,
            "place_id": 1,
            "src": "https://lh3.googleusercontent.com/QDwk6G6ZCwYzVH-yJHQ824GZ1wKW2XXus8hxxdH-4AAHz-FejC-UVydyE5le1ZvPTIo1cXw1G25qhljwS8dgyIedCCFtIxBonznVQ4LfaPDEZVWmqKS1vFqVUFFNmMmMlKQ5eCpfhNI=w2400",
            "title": "Atlas Pizza's play area",
            "user_id": 5,
          },
        ]
      }
      imagesClass="review-image"
    />
  </div>
</li>
`;
